### DISCUSSION TITLE
[RTPS_MSG_OUT Error] error: couldn't generate segment_name -> Function init

### DISCUSSION THREAD

**Discussion by genvictpeter** [eProsima/Fast-DDS#3853] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

this is the version information for using fastdds.
libfastcdr v1.1.0
libfastrtps v2.12.0
libfoonathan_memory-0.7.3
libtinyxml2 v9.0.0

### Current behavior

Every time we initialize fastdds Or run a long time to appear we get this Error,[RTPS_MSG_OUT Error] error: couldn't generate segment_name -> Function init

### Steps to reproduce

The cross tool chain used is arm-linux-gnueabihf- 

### Fast DDS version/commit

libfastcdr v1.1.0
libfastrtps v2.12.0 master
libfoonathan_memory-0.7.3
libtinyxml2 v9.0.0

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

Initial configuration, not modified

### XML configuration file

```xml
Initial configuration, not modified
```

### Relevant log output

```shell
Initial configuration, not modified
```

### Network traffic capture

Initial configuration, not modified

**Comment by genvictpeter** [comment]
![image](https://github.com/eProsima/Fast-DDS/assets/66452571/1f26962e-4cea-4653-b1fd-f16be2dae429)

**Comment by genvictpeter** [comment]
![image](https://github.com/eProsima/Fast-DDS/assets/66452571/89151027-2192-46b2-aa5e-1ad70c645f6b)
![image](https://github.com/eProsima/Fast-DDS/assets/66452571/dcf20ed0-6aef-443e-bdcf-66d54672ff80)

**Comment by JLBuenoLopez** [comment]
Thanks for the ticket @genvictpeter 

Can you provide more information about the platform you are using? Seeing the provided information you seem to be cross-compiling for an arm platform. You can find the officially supported platforms in [this document](https://github.com/eProsima/Fast-DDS/blob/master/RELEASE_SUPPORT.md). I am moving the issue to the Unofficial support discussion forum for the moment as per [Fast DDS Contributing](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md). In order for Shared Memory Transport or Data-sharing to work, Fast DDS should be able to create the segments in the shared memory. This depends on the platform being run as some platforms require additional permissions or use a different path for the shared memory files. Consequently, I am afraid maybe we are not going able to help with your specific use case. As a workaround, you can disable Shared Memory Transport (disabling the builtin transports and creating and enabling an UDP transport, for example. you can find more information is [Fast DDS documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/udp/udp.html#enabling-udp-transport)). The documentation also explains how to [disable Data-sharing](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/datasharing.html#data-sharing-delivery-kind).

You might also consider contacting [Fast DDS commercial support team](https://github.com/eProsima/Fast-DDS/tree/master#getting-help) to discuss options and support.

**Reply by genvictpeter** [reply]
I have appeared on both embedded arm linux systems, and the system information is shown in the picture below。
![image](https://github.com/eProsima/Fast-DDS/assets/66452571/3c95fb06-16d6-449f-99b7-99e780bd2d7d)
![image](https://github.com/eProsima/Fast-DDS/assets/66452571/339dd87d-507d-49bd-bddb-d9b5e8b75350)

**Reply by genvictpeter** [reply]
How does fastdds lose its high-speed advantage without using shared memory。

**Reply by JLBuenoLopez** [reply]
You can find some performance measurements in [this article](https://eprosima.com/index.php/resources-all/performance/eprosima-fast-dds-performance) from eProsima webpage.

**Reply by genvictpeter** [reply]
Sorry, I tried to create /dev/shm initialization on the device and it did not report, thank you for your reply.
![Uploading image.png…]()
